app/models/katello/concerns/pulp_database_unit.rb

```diff
@@ -20,8 +20,14 @@ def content_type
         self::CONTENT_TYPE
       end
 
+      def pulp3_support?
+        type = Katello::RepositoryTypeManager.find_repository_type content_type
+        type.pulp3_plugin && SmartProxy.pulp_master!.capabilities('Pulp3').try(:include?, type.pulp3_plugin)
+      end
+
       def content_unit_class
-        "::Katello::Pulp::#{self.name.demodulize}".constantize
+        version = pulp3_support? ? 3 : ""
+        "::Katello::Pulp#{version}::#{self.name.demodulize}".constantize
```

Comment:
can we change this to ask the smart proxy for the class (from the content type definition) rather than constructing it here?

Reply:
Updated..SmartProxy.pulp_master!.content_service(content_type) works fine..